src/client/QXmppUploadRequestManager.cpp

```diff
@@ -220,7 +220,8 @@ void QXmppUploadRequestManager::handleDiscoInfo(const QXmppDiscoveryIq &iq)
 
             // get size limit
             bool isFormNsCorrect = false;
-            for (const QXmppDataForm::Field &field : iq.form().fields()) {
+            const auto fields = iq.form().fields();
+            for (const QXmppDataForm::Field &field : fields) {
```

Comment:
This is a good catch. So, here `iq.form()` returns a copy of the internal field, which would be temporary living till the end of the expression and whose lifetime will not be extended. `fields()`, though, will choose the non-`const` overload, returning the reference to the field of the temporary, whose lifetime will not be extended as well. Am I correct in following this?

Reply:
I think we are hitting https://github.com/KDE/clazy/blob/master/docs/checks/README-detaching-temporary.md here.
I was initially looking for what causes this asan error, and I think it's likely this line.
```
SUMMARY: AddressSanitizer: heap-use-after-free /usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:178:32 in QList<QXmppDataForm::Field>::detach()
```